### PR TITLE
Feature/aba/by suchen/admin3

### PR DESCRIPTION
**Pull request by SuchenShi**
pull request

### CODE REVIEW

**Comment by ardithoxha** [comment]
Hey Suchen, I reviewed your code as well and it works. It might not be necessary (for now) but I liked that you implemented the router because you got to practice it and it gave us a chance to see it in action. I also like that you implemented the user interface and making the delete function update the actual database. I noticed you have some styling in the app.component.css and I recommend that you put them in the admin.componnent.css as those styles affect the elements in there. You also had some console logs left behind. I had mixed feeling about the userlist component. On one hand, I liked that you broke it down because it gave you a chance to practice component breakdown and emitting data. On the other hand, it made your code a little larger and a little more complicated than it could be.

**Comment by jqj5405** [comment]
Wow, I was surprised when my /users .json was deleted when testing your code ahah. For the most part, your code functions as expected. The use of a <router-outlet> was a nice surprise, I didn't expect to see it used yet. I noticed that userlist functions as a button for admin.component.ts, and thus can be combined with userlist.component.ts to simplify the use of @input/@output. The folders that are left will be userlist and userinfo as a child component, and admin.component.ts is not used and should be deleted. I've listed a few things that might be of interest below.

File: TeamProject/src/app/admin/userinfo/userinfo.component.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.userinfo_container{
+    align-items: center;
+}
```
Comment (jqj5405) [inline TeamProject/src/app/admin/userinfo/userinfo.component.css:3]: line 2 => inactive; requires display: flex or grid

File: TeamProject/src/app/admin/userinfo/userinfo.component.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit, EventEmitter, Input, Output } from '@angular/core';
+import { User } from '../interface/user.model';
+
+@Component({
+  selector: 'app-userinfo',
+  templateUrl: './userinfo.component.html',
+  styleUrls: ['./userinfo.component.css']
+})
+export class UserinfoComponent implements OnInit {
+  @Input() userinfo!: User;
```
Comment (jqj5405) [inline TeamProject/src/app/admin/userinfo/userinfo.component.ts:10]: line 10 => use of ! is not required

File: TeamProject/src/app/admin/userlist/userlist.component.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { User } from '../interface/user.model';
+
+@Component({
+  selector: 'app-userlist',
+  templateUrl: './userlist.component.html',
+  styleUrls: ['./userlist.component.css'],
+})
+export class UserlistComponent implements OnInit {
+  @Input() userinfo!: User;
```
Comment (jqj5405) [inline TeamProject/src/app/admin/userlist/userlist.component.ts:10]: line 10 => use of ! is not required

File: TeamProject/src/app/admin/admin.component.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+@Component({
+  selector: 'app-admin',
+  templateUrl: './admin.component.html',
+  styleUrls: ['./admin.component.css']
+})
+export class AdminComponent implements OnInit {
+  userlist!: User[];
+  FocusUser!: User;
+  FocusId!:number;
+  searchUser!: User[];
```
Comment (jqj5405) [inline TeamProject/src/app/admin/admin.component.ts:14]: line 11-14 => use of ! is not required

File: TeamProject/src/app/admin/admin.component.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  ngOnInit(): void {
+    this.userListService.getUsers().subscribe((data: any) => {
+      this.userlist = data;
+      console.log(this.userlist);
+    });
+  }
+  getFocus(focusid: number) {
+    this.FocusUser = this.userlist.filter((search) => {return search.id == focusid;})[0];
+    console.log(this.FocusUser);
+    this.FocusId = this.FocusUser.id;
```
Comment (jqj5405) [inline TeamProject/src/app/admin/admin.component.ts:27]: line 27 => no need to retrieve id from FocusUser, retrieve from focusid param instead
this.FocusId = this.FocusUser.id;
this.FocusID = focusid;

File: TeamProject/src/app/admin/admin.component.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  constructor(private userListService: UserlistService) {}
+
+  ngOnInit(): void {
+    this.userListService.getUsers().subscribe((data: any) => {
+      this.userlist = data;
+      console.log(this.userlist);
+    });
+  }
+  getFocus(focusid: number) {
+    this.FocusUser = this.userlist.filter((search) => {return search.id == focusid;})[0];
```
Comment (jqj5405) [inline TeamProject/src/app/admin/admin.component.ts:25]: line 25, 32 => (f.y.i) 	()=>{return} 	// result
						()=>			// result
						()=>{} 			// undefined

File: TeamProject/src/app/admin/admin.component.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    console.log(this.FocusUser);
+    this.FocusId = this.FocusUser.id;
+    console.log('Focusid: ',this.FocusId);
+  }
+  deleteUser(id:number) {
+    this.userListService.deleteUser(id).subscribe(()=>{
```
Comment (jqj5405) [inline TeamProject/src/app/admin/admin.component.ts:31]: line 31 => (f.y.i) if deletion from remote is not wanted, remove and keep line 32

**Review by jqj5405 — COMMENTED** [review]
Suggestions

**Pull request closed without merging**